src/deployment.rs

```diff
@@ -102,7 +220,13 @@ impl Artifact {
 }
 
 impl Deployment {
-    pub fn new( package: &CargoPackage, target: &CargoTarget, result: &CargoResult ) -> Result< Self, Error > {
+    pub fn new(
+            package: &CargoPackage,
+            target: &CargoTarget,
+            result: &CargoResult,
+            mut with_serve_path: Option<DeployWithServePath>
```

Comment:
Please make this function take `serve_path: String` instead of `with_serve_path`. There is no point in making anything here `Optional` since both `deploy_path` and `serve_path` have well defined defaults we can use (both being `/`), so it makes no sense to overcomplicate the code.

Reply:
tldr:
It is complicated because I want to check if we need to process the `js contents` or not. If `serve_path` is default or there is no `.wasm` file: do nothing on `js contents`, else, must process it. Another reason is I am too *lazy* (:-1: to myself)  to add more code (than a `None`) to `cmd_start.rs`.

Long story: (Please correct me if I say something not correct, and apologies for my poor English)

This is because of the involvement in `cmd_start.rs` (line 146), 
```rust
let deployment = Deployment::new( &package, &target, &result, None )?;
```
If don't want to take an `Option`, then it will become (`None` would be better?)
```rust
let config = project.get_default....
let deployment = Deployment::new( &package, &target, &result, config )?;
```
I think it is ok with that. But there is a more important thing to consider is that we must **process the `contents`** of the `js` file. If I understand correctly, we do not need to process it in case of `asmjs`, is this right? (don't know this exactly because I do not use `asmjs`) (and do not process the `js` file if `serve-path` is default) So, I add `DeployWithServePath` to monitor if there is a `.wasm` file and to store its **actual name** (on the filesystem, and that is also use in the `js`). Then when process the `js contents`, we check if it is required or not, if not, just do nothing. Otherwise, find the `.wasm` filename (recorded by `DeployWithServePath`) in js file, insert `serve-path`.

If you want to pass `serve-path` as  a `String`, then we need a second parameter that indicate if `target` is a `wasm32` or not (If it is not, we do not process the `js contents`), and it will become something like this:
```rust
    pub fn new(
            package: &CargoPackage,
            target: &CargoTarget,
            result: &CargoResult,
            serve_path: String,      // this
            target_is_wasm: bool,    // and this
        ) -> Result< Self, Error >
```
Finally, with the `fn` declaration above, we check if it is default or not by comparing `serve_path.len() == 0` to know if it is needed to process or not process the `js contents`. If we pass as `Option<String>` it is clearer if we are in the default case or not.

Reply:
What I tried is actually *hacking* the `.js` file :-1:! The reason is that I don't want to touch any code outside of the `deploy` process. I think that putting any hacky things into `DeployWithServePath` then if we have full support for `serve-path` in both `wasm32-unknown-emscripten` and `wasm32-unknown-unknown` just remove it all. 

But I understand that it is not good for the consistency of the whole project! I just don't have much time to dive deeper into build process and especially into Emscripten right now (even you have to say "*I don't know*"). So, let this halted here for awhile, until I have some free time to devote more into this.

Reply:
Well, initially I would be fine with just doing a search and replace for `wasm32-unknown-emscripten`; as long as it works we can always fix it later. The initial version of this doesn't have to be perfect, but I'm willing to accept only minor hacks as opposed to major ones. (:

Sure, no problem, thanks for investing the time into this. (: